spec/basic_examples_spec.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+def expect_failure(&block)
+  expect(&block).to raise_exception(RSpec::Expectations::ExpectationNotMetError)
+end
+
+RSpec.describe 'some basic hypothesis tests' do
+  it 'should think integer addition is commutative' do
+    hypothesis do
+      x = given integers
```

Comment:
Will this work with [let-syntax](https://relishapp.com/rspec/rspec-core/v/2-11/docs/helper-methods/let-and-let)?

Reply:
I don't think so, but I'm not that familiar with let-syntax so I'm not sure what you have in mind. Essentially the hypothesis block will just be repeatedly run with different outputs from these values - you can use let-syntax if you want, but it will be cached for the entire test, not a different value per block.

Reply:
It'll work.

let(:value) { x } creates a local method called value which evaluates the contents of the block the first time, and then memoizes the result for the duration of the example. I don't think it's going to work if you re-use the reference, however:

```ruby
# this won't work
let(:stream) { given integers}

it "finds addition is commutative" do
  x = stream
  y = stream

  expect(x + y).to eq(y + x)  
end
```

however, I'd expect all examples @DRMacIver gives in his example spec to work with `let`